deepcell_spots/applications/spot_detection.py

```diff
@@ -0,0 +1,252 @@
+        # Create the application
+        app = SpotDetection()
+        # Find spot locations
+        coords = app.predict(im)
+    Args:
+        model (tf.keras.Model): The model to load. If ``None``,
+            a pre-trained model will be downloaded.
+    """
+
+    #: Metadata for the dataset used to train the model
+    dataset_metadata = {
+        'name': 'general_train',
+        'other': """Pooled FISH data including MERFISH data
+                    and SunTag viral RNA data"""
+    }
+
+    #: Metadata for the model and training process
+    model_metadata = {
+        'batch_size': 1,
+        'lr': 0.01,
+        'lr_decay': 0.99,
+        'training_seed': 0,
+        'n_epochs': 10,
+        'training_steps_per_epoch': 552
+    }
+
+    def __init__(self, model=None):
+
+        if model is None:
+            archive_path = tf.keras.utils.get_file(
+                'SpotDetection.tgz', MODEL_PATH,
+                file_hash='2b9a46087b25e9aab20a2c9f67f4f559',
+                extract=True, cache_subdir='models'
+            )
+            model_path = os.path.splitext(archive_path)[0]
+            model = tf.keras.models.load_model(
+                model_path, custom_objects={
+                    'regression_loss': DotNetLosses.regression_loss,
+                    'classification_loss': DotNetLosses.classification_loss
+                }
+            )
+
+        super(SpotDetection, self).__init__(
+            model,
+            model_image_shape=model.input_shape[1:],
+            model_mpp=0.1,
+            preprocessing_fn=min_max_normalize,
+            postprocessing_fn=y_annotations_to_point_list_max,
+            dataset_metadata=self.dataset_metadata,
+            model_metadata=self.model_metadata)
+
+    def _postprocess(self, image, **kwargs):
+        """Applies postprocessing function to image if one has been defined.
+        Differs from parent class in that it returns a set of coordinate spot
+        locations, so handling of dimensions differs.
+
+        Otherwise returns unmodified image.
+        Args:
+            image (numpy.array or list): Input to postprocessing function
+                either an ``numpy.array`` or list of ``numpy.arrays``.
+        Returns:
+            list: coordinate spot locations
+        """
+        if self.postprocessing_fn is not None:
+            t = timeit.default_timer()
+            self.logger.debug('Post-processing results with %s and kwargs: %s',
+                              self.postprocessing_fn.__name__, kwargs)
+
+            output = self.postprocessing_fn(image, **kwargs)
+
+            self.logger.debug('Post-processed results with %s in %s s',
+                              self.postprocessing_fn.__name__,
+                              timeit.default_timer() - t)
+
+        elif isinstance(image, list) and len(image) == 1:
+            output = image[0]
+
+        return output
+
+    def _predict_segmentation(self,
```

Comment:
I'm pretty sure you could change this function name to `_predict` without causing any problems for the application